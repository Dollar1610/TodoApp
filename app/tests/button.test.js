import React from 'react';
import { shallow } from 'enzyme';
import Button from './../body/components/button';


describe('<Button />', () => {
  it('renders a <button>', () => {
    const renderedComponent = shallow(
      <Button />
    );
    expect(
      renderedComponent.find('button').node
    ).toBeDefined();
  });
  it('should render a button title', () => {
    const wrapper = shallow(
      <Button buttonName={'Title'} />
    );
    expect(wrapper.props().children).toEqual('Title');
  });
  it('handles clicks', () => {
    const onClickSpy = jest.fn();
    const renderedComponent = shallow(<Button event={onClickSpy} />);
    renderedComponent.find('button').simulate('click');
    expect(onClickSpy).toHaveBeenCalled();
    expect(onClickSpy).toMatchSnapshot();
  });
});
