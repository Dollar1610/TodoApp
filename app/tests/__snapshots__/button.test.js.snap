// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> handles clicks 1`] = `[Function]`;
